### ISSUE TITLE
rsync-version not picket up from alias file

### ISSUE THREAD

**Issue by hanoii** [drush-ops/drush#758]
The example.drushrc.php file says this:

```php
/**
 * Drush requires at least rsync version 2.6.4 for some functions to work
 * correctly.  rsync version 2.6.8 or earlier may give the following error
 * message: "--remove-source-files: unknown option".  To fix this, set
 * $options['rsync-version'] = '2.6.8'; (replace with the lowest version of
 * rsync installed on any system you are using with Drush).  Note that this
 * option can also be set in a site alias, which is the preferred solution if
 * newer versions of rsync are available on some of the systems you use.
 * See: http://drupal.org/node/955092
 */
```

I have a live server with rsync 2.6.8 and I tried unsuccessfully to add it to the live alias. I tried adding it to dev (the other alias) as well and nothing.

I added it to the $alias['live'] directory, then to specific-comments key, then also just as plain $options['rsync-version'] and nothing. The only way this worked for me was adding it to a drushrc.php file, but that makes it global to drush.

Any ideas?

**Comment by frederickjh** [comment]
Hi!
I am on drush version 7.0.0 and **--rsync-version** does not even work on the commandline with **sql-sync**. So I am not surprised that it does not work in an alias file. Looks like this was only implemented for **drush rsync**.  Is it possible to get this flag added for **sql-sync** on drush 7.x and in the alias  file?

I am facing a problem on a server that is running **rsync version 2.6.8 protocol version 29**. **drush rsync** works but **drush sql-sync** does not. I think that this is do to the fact that the flag **--remove-source-files** that drush uses(rsync protocol 30) was **--remove-sent-files** in protocol 29.
See this issue on d.o for more info [drush support for rsync version lower than 2.6.9](https://www.drupal.org/node/955092)

Thanks!

**Comment by greg-1-anderson** [comment]
rsync uses strict option handling (see `drush topic strict`), so you need to make sure you are using `drush --rsync-version=... rsync ...` rather than `drush rsync --rsync-version=...`.

It's been quite a few years since I've needed the 2.6.8 protocol, though, so I have not tested this recently. I think five years ago it was working fine with sql-sync, with the flag being set via an alias file. If that isn't working any longer, patches would be welcome.

**Comment by hanoii** [comment]
I just entered a pulled request as I faced a similar issue again. I changed two things, one is to allow rsync-version to be passed on to sql-sync.

The other is to support that in the sql-sync command, as that's was not taken rsync-version into account when adding --remove-source-files.

This now works by either passing on --rsync-version=2.6.8 to sql-sync, putting it on drushrc.php or adding it to 

'source-command-specific' (something weird - this also worked without the added option to sql-sync)

Now, something that's odd, which I tried to debug but I couldn't:

If I add the following to the alias file:

```php
  'source-command-specific' => array (
    'sql-sync' => array (
      'rsync-version' => '2.6.8',
    ),
  ),
```

It works (with or without the addition of --rsync-version to sql-sync but with the other bit of the pull request).

But if I add it like this:

```php
  'command-specific' => array (
    'sql-sync' => array (
      'rsync-version' => '2.6.8',
    ),
  ),
```

It doesn't.
